src/ElectronBackend/main/listeners.ts

```diff
@@ -103,43 +103,130 @@ async function writeOutputJsonToFile(
 
 export function getOpenFileListener(
   mainWindow: BrowserWindow,
+  onOpen: () => void,
 ): () => Promise<void> {
-  return createListenerCallbackWithErrorHandling(mainWindow, async () => {
-    const filePaths = openFileDialog();
+  return createVoidListenerCallbackWithErrorHandling(mainWindow, async () => {
+    const filePaths = openOpossumFileDialog();
     if (!filePaths || filePaths.length < 1) {
       return;
     }
-    let filePath = filePaths[0];
+    const filePath = filePaths[0];
 
-    if (filePath.endsWith(outputFileEnding)) {
-      filePath = tryToGetInputFileFromOutputFile(filePath);
-    }
-
-    await handleOpeningFile(mainWindow, filePath);
+    await handleOpeningFile(mainWindow, filePath, onOpen);
   });
 }
 
 export async function handleOpeningFile(
   mainWindow: BrowserWindow,
   filePath: string,
+  onOpen: () => void,
 ): Promise<void> {
-  const isOpossumFormat = isOpossumFileFormat(filePath);
   logger.info('Initializing global backend state');
-  initializeGlobalBackendState(filePath, isOpossumFormat);
+  initializeGlobalBackendState(filePath, true);
 
-  if (!isOpossumFormat) {
-    const dotOpossumFilePath = getDotOpossumFilePath(filePath);
-    if (fs.existsSync(dotOpossumFilePath)) {
-      initializeGlobalBackendState(dotOpossumFilePath, !isOpossumFormat);
-    }
-    mainWindow.webContents.send(AllowedFrontendChannels.ShowFileSupportPopup, {
-      showFileSupportPopup: true,
-      dotOpossumFileAlreadyExists: fs.existsSync(dotOpossumFilePath),
-    });
-    return;
-  }
+  await openFile(mainWindow, filePath, onOpen);
+}
+
+export function getImportFileListener(
+  mainWindow: BrowserWindow,
+  fileFormat: FileFormatInfo,
+): () => Promise<void> {
+  return createVoidListenerCallbackWithErrorHandling(mainWindow, () => {
+    mainWindow.webContents.send(
+      AllowedFrontendChannels.ImportFileShowDialog,
+      fileFormat,
+    );
+  });
+}
+
+export function getImportFileSelectInputListener(
+  mainWindow: BrowserWindow,
+): (
+  _: Electron.IpcMainInvokeEvent,
+  fileFormat: FileFormatInfo,
+) => Promise<string | null> {
+  return createListenerCallbackWithErrorHandling(
+    mainWindow,
+    (_: Electron.IpcMainInvokeEvent, fileFormat: FileFormatInfo) => {
+      const filePaths = openNonOpossumFileDialog(fileFormat);
+      if (!filePaths || filePaths.length < 1) {
```

Comment:
#nit an array won't have negative length
```suggestion
      if (!filePaths?.length) {
```

Reply:
resolved